autoblocks/_impl/tracer.py

```diff
@@ -1,24 +1,55 @@
 import asyncio
+import atexit
+import contextvars
+import dataclasses
+import inspect
 import logging
+import time
 import uuid
 from contextlib import contextmanager
-from dataclasses import dataclass
 from datetime import datetime
 from datetime import timedelta
 from datetime import timezone
 from typing import Any
 from typing import Dict
 from typing import Generator
+from typing import List
 from typing import Optional
 
 from autoblocks._impl import global_state
 from autoblocks._impl.config.constants import INGESTION_ENDPOINT
+from autoblocks._impl.testing.models import BaseEventEvaluator
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.testing.models import TracerEvent
+from autoblocks._impl.util import SEND_EVENT_CORO_NAME
 from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks._impl.util import gather_with_max_concurrency
 
 log = logging.getLogger(__name__)
 
 
-@dataclass
+@atexit.register
+def _cleanup_tracer() -> None:
```

Comment:
nit: you don't need underscores in this file because it's in the `_impl/` folder, so you don't need to worry about it being exposed to the end user